src/lib/entry/ServiceTestFixture.test.ts

```diff
@@ -0,0 +1,117 @@
+import {exportDependency, PluginConfig} from '../core/PluginConfig'
+import {ServiceTestFixture} from './ServiceTestFixture'
+import {spyService, SpyService} from '../core/mocks/SpyService'
+import {expect} from 'chai'
+import {SinonSpy, spy} from 'sinon'
+import {inject, Service} from '../../index'
+
+describe('ServiceTestFixture', () => {
+  class TestPlugin extends PluginConfig {
+    @exportDependency('value')
+    getValue = () => this.value
+
+    value = 1
+  }
+
+  it('Service can access an inject value', async () => {
+    class ServiceWithInjectedValue extends Service {
+      @inject('value')
+      value: () => number
+    }
+
+    const fixture = await ServiceTestFixture.createMountedService<ServiceWithInjectedValue>({
+      service: ServiceWithInjectedValue,
+      plugins: [TestPlugin]
+    })
+
+    expect(fixture.service.value()).to.eql(1)
+  })
+
+  it('Service can access the value of the plugin', async () => {
+    const fixture = await ServiceTestFixture.createMountedService({
+      service: SpyService,
+      plugins: [TestPlugin]
+    })
+
+    expect(fixture.service.context['@appContext'].getInjectedObject('value')()).to.eql(1)
+  })
+
+  it('allows plugins to be accessed and stubbed', async () => {
+    const fixture = await ServiceTestFixture.createMountedService({
+      service: SpyService,
+      plugins: [TestPlugin],
+    })
+
+    fixture.getPlugin(TestPlugin).value = 5
+
+    expect(fixture.service.context['@appContext'].getInjectedObject('value')()).to.eql(5)
+  })
+
+  describe('setState', () => {
+    it('should return value afterwards', async () => {
+      const fixture = await ServiceTestFixture.createMountedService({
+        service: SpyService,
+      })
+
+      const STATE_KEY = 'key'
+      const STATE_VALUE = 'value'
+
+      fixture.service.setState({[STATE_KEY]: STATE_VALUE})
+
+      expect(fixture.service.state[STATE_KEY]).to.eql(STATE_VALUE)
+    })
+
+    it('calls supplied callback', async () => {
+      const fixture = await ServiceTestFixture.createMountedService({
+        service: SpyService,
+      })
+
+      const STATE_KEY = 'key'
+      const STATE_VALUE = 'value'
```

Comment:
Can these be extracted so that they're not duplicated in the tests?

Reply:
Yes they can